lib/src/components/sprite_component.dart

```diff
@@ -24,11 +24,18 @@ class SpriteComponent extends PositionComponent {
   Paint overridePaint;
 
   /// Creates a component with an empty sprite which can be set later
-  SpriteComponent();
-
-  SpriteComponent.fromImage(Vector2 size, Image image)
-      : this.fromSprite(size, Sprite(image));
-
+  SpriteComponent({
+    Vector2 position,
+    Vector2 size,
+    this.sprite,
+    this.overridePaint,
+  }) : super(position: position, size: size);
+
+  @Deprecated('Use SpriteComponent instead')
+  factory SpriteComponent.fromImage(Vector2 size, Image image) =>
```

Comment:
This factory still makes sense, remove the deprecation and add `size` and `position` as optional

Reply:
added